File: Source/Data/plugins/skill/clickitem.py
from server.util import Plugin
from server.model.players.skills.herblore import Cleaning
from server.model.players.skills.cooking import Food

def click(player, itemId, itemSlot):
			Cleaning.handleCleaning(player, itemId, itemSlot)

# Eating food types like Sharks, etc..
def click(player, itemId, itemSlot):
		if (Food.FoodToEat.forId(itemId)):
			Food.eat(player, itemId, itemSlot)

File: Source/src/server/model/players/packet/impl/ClickItem.java
package server.model.players.packet.impl;

import server.model.players.Player;
import server.model.players.packet.PacketType;
import server.util.Plugin;

/**
 * Clicking an item, bury bone, eat food etc
 **/
public class ClickItem implements PacketType {

	@Override
	public void processPacket(Player player, int packetType, int packetSize) {
		@SuppressWarnings("unused")
		int junk = player.getInStream().readSignedWordBigEndianA();
		int itemSlot = player.getInStream().readUnsignedWordA();
		int itemId = player.getInStream().readUnsignedWordBigEndian();
		if (itemId != player.playerItems[itemSlot] - 1) {
			return;
		}
		/**
		 * Gets the static usage from the skill class and access it in the Python script
		 */
		Plugin.execute("click", player, itemId, itemSlot);
		
		Plugin.execute("itemClick_" + itemId, player, itemId, itemSlot);
	}
}